### PR TITLE
add sentry app

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Add sentry to this app

Decided to add the app name to the environment. This means we can have the same project for all back end micro services and then use the environment to help filter between different micro services. 

#122 

## How Has This Been Tested?

CI tests + tested it locally and got error

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet_app/app.py [hunk 1]
```diff
@@ -55,6 +56,14 @@
     preprocess_nwp_data,
 )
 from pvnet_app.forecast_compiler import ForecastCompiler
+from pvnet_app.sentry import traces_sampler
+
+# sentry
+sentry_sdk.init(
+    dsn=os.getenv("SENTRY_DSN", ""),
+    environment=f'{os.getenv("ENVIRONMENT", "development")}__uk-pvnet-app--{pvnet_app.__version__}',
```
Comment (peterdudfield) [inline pvnet_app/app.py:64]: i think ill move the version out of this

File: pvnet_app/app.py [hunk 2]
```diff
@@ -55,6 +56,14 @@
     preprocess_nwp_data,
 )
 from pvnet_app.forecast_compiler import ForecastCompiler
+from pvnet_app.sentry import traces_sampler
+
+# sentry
+sentry_sdk.init(
+    dsn=os.getenv("SENTRY_DSN", ""),
+    environment=f'{os.getenv("ENVIRONMENT", "development")}__uk-pvnet-app',
```
Comment (peterdudfield) [inline pvnet_app/app.py:64]: update to local

File: pvnet_app/sentry.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+def traces_sampler(sampling_context):
+    """
+    Filter tracing for sentry logs.
+
+    Examine provided context data (including parent decision, if any)
+    along with anything in the global namespace to compute the sample rate
+    or sampling decision for this transaction
+    """
+
+    if os.getenv("ENVIRONMENT", "local") == "local":
```
Comment (braddf) [inline pvnet_app/sentry.py:14]: Purely as an FYI as I looked at this at some point when adding it, possibly to Quartz Energy – not sure this is actually doing what we think it is / would like it to, but we can revisit another time for sure as it also didn't seem to be doing any harm as it is

File: pvnet_app/app.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (braddf) [reply pvnet_app/app.py]: I think I would propose one of two ways of proceeding:
1. Keep this as intended by default, and have 1 project mapped to 1 Github repo – if it did get unfeasibly cluttered, we could also split by Team and have a team for Product and a team for Forecast stuff (so yeah kind of "customer-facing" and "backend", but I think backend is a bit broad);
2. Try merging the projects with 2-3 and see how the Sentry GH integration, source code stuff and general UI with project vs. environment behaves, and then assess which way makes more sense.

**Pull request merged**